### ISSUE TITLE
Removal of Semicolon from Base Class Node2

### ISSUE THREAD

**Issue by Shimul-Baidya** [Shimul-Baidya/MBDyn#12]
In [79b2098](https://github.com/Shimul-Baidya/MBDyn/commit/79b20989f4491a8aa00f35952bf993faebea7e53), I removed the semicolon (;) from the __str__ method of the base class Node2 to address a specific issue: when adding accelerations to dynamic and modal nodes, the semicolon at the end of the base class output caused duplicates. This made it challenging to append the accelerations parameter correctly since the base class already included a semicolon.

Instead I added semicolons separately in the subclasses.

***Previously:***
https://github.com/Shimul-Baidya/MBDyn/blob/79b20989f4491a8aa00f35952bf993faebea7e53/contrib/PythonPreprocessor/MBDynLib.py#L611-L622

***Now:***
https://github.com/Shimul-Baidya/MBDyn/blob/79b20989f4491a8aa00f35952bf993faebea7e53/contrib/PythonPreprocessor/MBDynLib.py#L670-L680

Is it correct to remove the semicolon from the base class Node2 to resolve this issue?
Will removing the semicolon cause problems if someone creates a node directly from the base Node2 class?

**Comment by Shimul-Baidya** [comment]
Now that I think of it, won't it work if I included `accelerations` in the base `Node2` class as an optional parameter and don't remove the semicolon(;) from the base `Node2` class and do not add semicolon in the subclasses separately?

**Comment by zanoni-mbdyn** [comment]
Hi @Shimul-Baidya, 

if, as it seems to me, the `Node2` class is not meant to be used to define a structural node, then it is fine to conclude the definition only in the derived classes, since they are the ones that will be used anyway. But we should make sure that this is the case: there must be a derived class for _each_ type of `structural node`. 

And we should warn (or prevent, if there is a way... @Maarrk?) the user that is attempting to call the `Node2` constructor directly, and not through the constructor of one of the derived classes.

P.S. by the way, `StructuralNode` would be a better name for the `Node` class imo, since we also have hydraulic nodes, thermal nodes and electrical nodes...

**Comment by Maarrk** [comment]
> And we should warn (or prevent, if there is a way... @Maarrk?) the user that is attempting to call the `Node2` constructor directly, and not through the constructor of one of the derived classes.

Yes, we are already enforcing that in e.g. `ConstitutiveLaw` through abstract classes. `MBEntity` has `__str__` marked as abstract, meaning you can only create instances of subclasses which override it:
https://github.com/Shimul-Baidya/MBDyn/blob/aec904ea1f3655d6e8232cbb01869a487eb82c96/contrib/PythonPreprocessor/MBDynLib.py#L91-L97

The definition here doesn't define the `__str__` method, but every subclass for specific law does.
https://github.com/Shimul-Baidya/MBDyn/blob/aec904ea1f3655d6e8232cbb01869a487eb82c96/contrib/PythonPreprocessor/MBDynLib.py#L5124-L5126

https://github.com/Shimul-Baidya/MBDyn/blob/aec904ea1f3655d6e8232cbb01869a487eb82c96/contrib/PythonPreprocessor/MBDynLib.py#L5178-L5193

We even have a test for that:
https://github.com/Shimul-Baidya/MBDyn/blob/aec904ea1f3655d6e8232cbb01869a487eb82c96/contrib/PythonPreprocessor/test_MBDynLib.py#L81-L86
